src/test/groovy/com/intuit/graphql/orchestrator/integration/EntityWithNestedFieldResolverSpec.groovy

```diff
@@ -0,0 +1,136 @@
+package com.intuit.graphql.orchestrator.integration
+
+import com.intuit.graphql.orchestrator.GraphQLOrchestrator
+import com.intuit.graphql.orchestrator.ServiceProvider
+import graphql.ExecutionInput
+import graphql.ExecutionResult
+import graphql.schema.GraphQLSchema
+import helpers.BaseIntegrationTestSpecification
+import spock.lang.Subject
+
+class EntityWithNestedFieldResolverSpec extends BaseIntegrationTestSpecification {
+
+    def schemaA = """   
+        type Query {
+            arootField: ARootType
+        }
+
+        type ARootType {
+            aTopField(p1: String!): AObjectType!
+        }
+        
+        type BObjectType @key(fields: "bObjectField") @extends {
+            bObjectField: String @external 
+            aTopField: AObjectType          
+        }        
+        
+        type AObjectType {
+            aObjectField: String
+            cTopField: CObjectType @resolver(field: "cTopField")
+        }   
+        
+        type CObjectType
+        
+        directive @resolver(field: String!, arguments: [A_ResolverArgument!]) on FIELD_DEFINITION
+        input A_ResolverArgument { # If this is type, dapi does not throws error
+            name : String!
+            value : String!
+        }         
+    """
+
+    def QUERY_A = "query (\$REPRESENTATIONS:[_Any!]!) {_entities(representations:\$REPRESENTATIONS) {... on BObjectType {aTopField {aObjectField __typename}}}}"
```

Comment:
This is the actual query but currently will not match.   Need to add capability for EntityFetcherBatchLoader to remove external fields.